### PR TITLE
Creating a data model

### PR DESCRIPTION
**Pull request by rica213**
In this  PR, here are the things I have done:
- Set up database to my project
- Use migration file to create my tables
- Generate the related model class for each tables
- Define the associations between tables
- Add index to all foreign keys
- Generate the schema file
- Update readme file

### CODE REVIEW

File: db/migrate/20230511214144_create_comments.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
+
+      t.timestamps
+    end
```
Comment (Gambit142) [inline db/migrate/20230511214144_create_comments.rb:9]: - [ ] Kindly change the attribute that references the `user` table from `:user` to `:author` based on the requirements. It should look like this:

```ruby
t.references :author, foreign_key: { to_table: 'users' }
```

Kindly implement this change 👍 😎

File: db/migrate/20230511214148_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
```
Comment (Gambit142) [inline db/migrate/20230511214148_create_likes.rb:9]: - [ ] Kindly change the attribute that references the `user` table from `:user` to `:author` based on the requirements. It should look like this:

```ruby
t.references :author, foreign_key: { to_table: 'users' }
```

Kindly implement this change 👍 😎

File: db/migrate/20230511190603_create_posts.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :author, null: false, foreign_key: { to_table: :users }
```
Comment (Gambit142) [inline db/migrate/20230511190603_create_posts.rb:4]: - [OPTIONAL] I would advise you to remove the `null: false` statement for this attribute. If you use it, you would always have to specify the `author_id` when creating a new instance of this model. But if you don't use it, you could easily create an instance of this model by passing the `User` model object 💻 🌵. For example,

#### With the `null: false`

`first_post = Post.create!(author_id: first_user.id, title: 'Hello', text: 'This is my first post')`

#### Without the `null: false`

`first_post = Post.create!(author: first_user, title: 'Hello', text: 'This is my first post')`

File: db/migrate/20230511214144_create_comments.rb (hunk 1 above)
Comment (Gambit142) [inline db/migrate/20230511214144_create_comments.rb:9]: - [OPTIONAL] I would advise you to remove the `null: false` statement for this attribute. If you use it, you would always have to specify the `author_id` when creating a new instance of this model. But if you don't use it, you could easily create an instance of this model by passing the `User` model object 💻 🌵. For example,

#### With the `null: false`

`first_post = Post.create!(author_id: first_user.id, title: 'Hello', text: 'This is my first post')`

#### Without the `null: false`

`first_post = Post.create!(author: first_user, title: 'Hello', text: 'This is my first post')`

File: db/migrate/20230511214148_create_likes.rb (hunk 2 above)
Comment (Gambit142) [inline db/migrate/20230511214148_create_likes.rb:9]: - [OPTIONAL] I would advise you to remove the `null: false` statement for this attribute. If you use it, you would always have to specify the `author_id` when creating a new instance of this model. But if you don't use it, you could easily create an instance of this model by passing the `User` model object 💻 🌵. For example,

#### With the `null: false`

`first_post = Post.create!(author_id: first_user.id, title: 'Hello', text: 'This is my first post')`

#### Without the `null: false`

`first_post = Post.create!(author: first_user, title: 'Hello', text: 'This is my first post')`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @rica213 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Created migration file for the `User` model ✔️ 
- Created migration file for the `Post` model ✔️
- Created migration file for the `Comment` model ✔️ 
- Created migration file for the `Like` model ✔️  

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider the comments tagged _**[OPTIONAL]**_ under the review summary. 👍 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: db/migrate/20230511214144_create_comments.rb (hunk 1 above)
Comment (rica213) [reply db/migrate/20230511214144_create_comments.rb:9]: Hi @Gambit142, thank you for the time to review my code. I appreciate it.
However, I'm a bit confused about this required change.
When you look closely at the ERD, only the post table uses author_id as the foreign key to reference the User table.
And the instruction is saying precisely, **_posts_**. Kindly, enlighten me on these required changes.

Screenshot from 2023-05-12 10-54-55


blog_app_erd_v1_1

File: db/migrate/20230511214148_create_likes.rb (hunk 2 above)
Comment (rica213) [reply db/migrate/20230511214148_create_likes.rb:9]: @Gambit142 Kindly reference to my previous comment

**Review by Davidosky007 — APPROVED** [review]
Hi @rica213 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Well written PR 👍 
* Good looking readme 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**